src/stores/licenseStore.ts

```diff
@@ -0,0 +1,125 @@
+import { create } from 'zustand'
+import supabase from '@/lib/integrations/supabase'
+import { devtools, subscribeWithSelector } from 'zustand/middleware'
+import { RealtimeChannel } from '@supabase/supabase-js'
+import { error as logError } from '@tauri-apps/plugin-log'
+
+export interface License {
+  id: string
+  status: string
+  license_type: 'perpetual' | 'subscription'
+  expiration_date: string | null
+}
+
+interface LicenseStoreState {
+  license: License | null
+  isLoading: boolean
+  error: Error | null
+  channel: RealtimeChannel | null
+  fetchLicense: (userId: string | null) => Promise<void>
+  initSubscription: (userId: string) => void
+  clearSubscription: () => Promise<void>
+}
+
+export const useLicenseStore = create<LicenseStoreState>()(
+  devtools(
```

Comment:
interesting, I need to look into this. I'm assuming this lets us inspect the state of the store from the chrome inspector?